contrib/assets_tutorial/pset_swap_tutorial.py

```diff
@@ -0,0 +1,334 @@
+#!/usr/bin/env python3
+
+from test_framework.authproxy import JSONRPCException
+from test_framework.daemon import Daemon, sync_all
+import argparse
+import sys
+from decimal import Decimal
+
+## PSET Swap Tutorial
+#
+# This script demonstrates how to implement an atomic swap, using a single
+# Elements transaction, between two parties, Alice and Carol. Alice has
+# ten thousand BTC and Carol has 1000 of a new asset. After the swap, Alice
+# will have the new asset and Carol will have the BTC.
+#
+# This script can be executed in a Python intepreter. The user is encouraged
+# to add tracing or debugging code throughout to better understand what is
+# going on.
+#
+
+## 0. Boilerplate to make the tutorial executable as a script
+#
+# Skip ahead to step 1 if you are reading
+#
+
+# Parse arguments
+parser = argparse.ArgumentParser()
+parser.add_argument("--elementsd-dir", type=str, default="./src")
+parser.add_argument("--no-cleanup", default=False, action="store_true")
+
+args = parser.parse_args()
+
+# Setup daemons
+alice = Daemon(
+    "Alice",
+    "elements",
+    args.elementsd_dir + "/elementsd",
+    "contrib/assets_tutorial/elements1.conf",
+    not args.no_cleanup,
+)
+
+carol = Daemon(
+    "Carol",
+    "elements",
+    args.elementsd_dir + "/elementsd",
+    "contrib/assets_tutorial/elements2.conf",
+    not args.no_cleanup,
+)
+
+## 1. Start nodes
+print ("1. Start nodes and setup scenario")
+
+# 1a. Turn on both nodes. Disable -validatepegin as we will just swap the
+#     initialcoins asset with a newly issued asset. Using the peg would
+#     be an inessential distraction.
+#
+alice.start(["-validatepegin=0"])
+carol.start(["-validatepegin=0"])
+alice.connect_to(carol)
+carol.connect_to(alice)
+alice.createwallet("wallet")
+carol.createwallet("wallet")
+alice.rescanblockchain()
+
+# 1b. Split the initial coins so that both sides have some (but Alice has more)
+alice.sendmany("", { alice.getnewaddress(): 10000 for x in range(0, 10) })
```

Comment:
To prevent `walletprocesspsbt` from filling in all of the inputs' data and then immediately blinding (which there is no way to prevent), basically jumping to the end of the tutorial in step 1.

Reply:
Oh, as for "why have a ton of smallish outputs", it was mainly to make coin selection more interesting, and also to avoid the "21M balance checks do not consider differing asset IDs" bug.